octodns/provider/ultra.py

```diff
@@ -287,7 +287,12 @@ def populate(self, zone, target=False, lenient=False):
                 name = zone.hostname_from_fqdn(record['ownerName'])
                 if record['rrtype'] == 'SOA (6)':
                     continue
-                _type = self.RECORDS_TO_TYPE[record['rrtype']]
+                try:
+                    _type = self.RECORDS_TO_TYPE[record['rrtype']]
+                except KeyError:
+                    self.log.warning('populate: ignoring record with '
+                                     'unsupported rrtype=%s', record)
```

Comment:
Seems like this will print out the `dict` of the record which for big/involved records might be noisy and hard to read. It may be preferable to do something like

```suggestion
                                     'unsupported rrtype, %s  %s', name, record['rrtype'])
```

Feel free to tweak or change that.